File: app/src/main/java/com/frama/miserend/hu/view/BottomNavigationViewBehavior.java
package com.frama.miserend.hu.view;

import android.content.Context;
import android.support.design.widget.AppBarLayout;
import android.support.design.widget.BottomNavigationView;
import android.support.design.widget.CoordinatorLayout;
import android.util.AttributeSet;
import android.view.View;

import com.frama.miserend.hu.utils.ViewUtils;

/**
 * Created by maczak on 2018. 02. 14..
 */

public class BottomNavigationViewBehavior extends CoordinatorLayout.Behavior<BottomNavigationView> {

    private int toolbarHeight;

    public BottomNavigationViewBehavior(Context context, AttributeSet attrs) {
        super(context, attrs);
        this.toolbarHeight = ViewUtils.getToolbarHeight(context);
    }

    @Override
    public boolean layoutDependsOn(CoordinatorLayout parent, BottomNavigationView fab, View dependency) {
        return dependency instanceof AppBarLayout;
    }

    @Override
    public boolean onDependentViewChanged(CoordinatorLayout parent, BottomNavigationView child, View dependency) {
        if (dependency instanceof AppBarLayout) {
            CoordinatorLayout.LayoutParams lp = (CoordinatorLayout.LayoutParams) child.getLayoutParams();
            int bottomMargin = lp.bottomMargin;
            int distanceToScroll = child.getHeight() + bottomMargin;
            float ratio = dependency.getY() / (float) toolbarHeight;
            child.setTranslationY(-distanceToScroll * ratio);
        }
        return true;
    }

}

File: app/src/main/java/com/frama/miserend/hu/home/view/BottomNavigationViewBehavior.kt
package com.frama.miserend.hu.home.view

import android.content.Context
import android.support.design.widget.AppBarLayout
import android.support.design.widget.BottomNavigationView
import android.support.design.widget.CoordinatorLayout
import android.util.AttributeSet
import android.view.View

import com.frama.miserend.hu.utils.ViewUtils
import com.frama.miserend.hu.utils.getToolbarHeight

/**
 * Created by maczak on 2018. 02. 14..
 */

class BottomNavigationViewBehavior(context: Context, attrs: AttributeSet) : CoordinatorLayout.Behavior<BottomNavigationView>(context, attrs) {

    private val toolbarHeight: Int = context.getToolbarHeight()

    override fun layoutDependsOn(parent: CoordinatorLayout?, fab: BottomNavigationView?, dependency: View?): Boolean {
        return dependency is AppBarLayout
    }

    override fun onDependentViewChanged(parent: CoordinatorLayout?, child: BottomNavigationView?, dependency: View?): Boolean {
        if (dependency is AppBarLayout) {
            val lp = child!!.layoutParams as CoordinatorLayout.LayoutParams
            val bottomMargin = lp.bottomMargin
            val distanceToScroll = child.height + bottomMargin
            val ratio = dependency.y / toolbarHeight.toFloat()
            child.translationY = -distanceToScroll * ratio
        }
        return true
    }

}